src/features/renameFileParts.ts

```diff
@@ -0,0 +1,208 @@
+import * as vscode from 'vscode'
+import { extname } from 'path-browserify'
+import { getActiveRegularEditor } from '@zardoy/vscode-utils'
+import { noCase, camelCase } from 'change-case'
+import { getExtensionContributionsPrefix, registerExtensionCommand, RegularCommands } from 'vscode-framework'
+import { proxy, subscribe } from 'valtio/vanilla'
+import { lowerCaseFirst } from 'lower-case-first'
+import { Utils as UriUtils } from 'vscode-uri'
+
+export const registerRenameFileParts = () => {
+    registerExtensionCommand('renameFileParts', async () => {
+        const activeEditor = getActiveRegularEditor()
+        if (!activeEditor) return
+
+        const { uri: sourceUri } = activeEditor.document
+        const fullFileName = UriUtils.basename(sourceUri)
+        let fileName = fullFileName
+        let fullExt = ''
+        // cut extension twice e.g. file.test.ts -> file & .test.ts
+        for (const _i of Array.from({ length: 2 })) {
+            const ext = extname(fileName)
+            if (ext.length === 0) continue
+
+            fileName = fileName.slice(0, -ext.length)
+            fullExt = `${ext}${fullExt}`
+        }
+
+        const quickPick = vscode.window.createQuickPick()
+
+        const parts = proxy([] as string[])
+        let isDisposeEnabled = true
+        /** expected: dash or underscore */
+        let separatorChar = ''
+
+        // init parts
+        noCase(fileName, {
```

Comment:
Didn't add file extension to parts as I thought it would be confusing for user and changing both file name and ext is extremely rare case to care about.